Changing css to relative sizing for index and earth

Ibexian opened the pull request:
So this is some of the changes I would do. I've only done them for the index page and the earth page, but since a lot of the css is reused in each of the files it'd be easy to reuse a lot of it. 

#### What I did:
1. Change the fixed-bg sizing to use 'vh' rather than a fixed pixel size
2. [Removed the absolution positioning and placement from the menu - in favor of margins and floats](https://github.com/shashaxdu/kaleidoscope/pull/1/files#diff-60543f47e2b3f5064797da20463ee979R30)
3. Used percents and 'vw' sizing to place the earth, moon, and sun
4. [Used flex format for the index buttons in the middle of the page](https://github.com/shashaxdu/kaleidoscope/pull/1/files#diff-bebe469df5fdb3f59ee8369d9857b3d5R41)

The above should allow for the pages to resize a bit better - though the sun's hover text is still a bit long for any screen smaller than 800px, so you may want to change the font size using [media queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries)

Commit:
Changing css to relative sizing for index and earth

shashaxdu commented:
@Ibexian Thanks so much, Will! Sorry it's taken a bit for me to respond. I will take a look at the edits with Manar.

Ibexian commented:
@shashaxdu OK - let me know if you two have questions or need something else ✨

shashaxdu committed:
objectplacement_fontsizing

I changed the placements of the left navbar, avatar/infinity objects, and bottom intro. I also changed fonts to em

shashaxdu commented:
@Ibexian Woohoo I made my first commit! Questions for you:

<img width="288" alt="screen shot 2017-09-09 at 11 38 55 pm" src="https://user-images.githubusercontent.com/13709857/30246772-3ac6ef32-95b8-11e7-888c-186df6f8b78d.png">
1) How do I make it so that on mobile, "Kaleidoscope" fits within the confines of the screen so that horizontal scrolling isn't necessary, and there's equal padding to the right and left? 

<img width="915" alt="screen shot 2017-09-09 at 11 42 25 pm" src="https://user-images.githubusercontent.com/13709857/30246785-a005ed94-95b8-11e7-9859-77f1cf9f01bf.png">
2) When I decrease the size of the browser like above, I want the avatar/infinity images to be equidistant between h2 and h3, even if that means the user has to scroll vertically. I played around with the 'position' but couldn't figure it out. 

I really want to combine all the CSS files but am dreading it.
Also, staying up late on a Sat. night working on this means I've now entered Next Level nerd status. 😂

shashaxdu committed:
CSS Merge: Earth -> Index

Merged CSS from Earth.css to IndividualtoInfinity.css (aka what will be Index.css)

shashaxdu commented:
Hi, @Ibexian!

(I deleted my last comment because the questions were no longer relevant after the latest commit.)

Questions for you:

1) I need your help with parallax 🙏🏻  Currently, the Headers and LeftNav move when vertically scrolling while everything else is static (see SolarSystem for example). However, it would make much more sense for only the images to move and for everything else to stay static. So for example, on the Index, the "Person" and "Infinity" images should move up and down as the user scrolls vertically while the background and everything else remain static. On the Earth page, the Earth, Moon, and Sun should move while the stars in the background remain static. This way, users will feel like by scrolling up and down, they are moving through space. 

2) Before, the "Person" and "Infinity" images on the Index page would increase in size with the hover function, but now it doesn't work and I can't figure out why?

3) This is a mobile view of the Index page. Is it possible to make everything fit horizontally within the screen so that horizontal scrolling isn't necessary (though it may be necessary on some other pages, such as the SolarSystem page.) 
<img width="303" alt="screen shot 2017-09-11 at 12 12 20 am" src="https://user-images.githubusercontent.com/13709857/30262643-8911fd5c-9686-11e7-83b0-841962982885.png">

4) Referring to the same image above, how can I either make the "Person" stack on top of the "Infinity" (in mobile view only) or make both of them in the same row but align vertically in the middle of the page rather than towards the top? 

Thank you!!! (Been working on this Sat. and Sun. night...next-level nerd status 👩🏻‍💻)

Commit:
Add hover resize

Commit:
Resize image height for mobile users on index

Ibexian commented:
Ok, looking at it now:
1) I can't actually scroll on the index page (ditto with the earth) so I'm not sure how to get the effect you are after...

[Scrolling on the index](https://william.kamovit.ch/img/indexScroll.gif)

2)  I've pushed a commit to get this effect - but the way you do it is by using a css 'pseudo-selector' called ':hover' to tell the style you want on hover. So I just added:

```
#person:hover {
  width: 16vmax;
}

#infinity:hover {
  width: 16vmax;
} 
```
3) Again, I'm not really seeing what you are seeing - are you on the correct branch? Here's what I'm seeing:
![Index when small](https://william.kamovit.ch/img/indexSmall.png)

4) As you can see by the above image - I made the images align vertically - the way this is done is something called 'media-queries' - essentially telling the browser to use different style based on the size of the screen. The code became something like this:

```
#images {
  height: 80vh;
}
@media (min-width: 600px) {
  #images {
    height: 50vh;
  }
}
```
Essentially - #images has a height of 80vh unless the screen is bigger than 600px (so anything bigger than a phone/small tablet)

I think it makes sense to merge this pull request in to master, so that it'll be easier to work on going forward (no branch confusion if there is one) - and we can open up new PRs for new changes.

shashaxdu committed:
CSS Merge: Avatar -> Index

Merged CSS from Avatar.css to IndividualtoInfinity.css and made small
changes, i.e. passport color/opacity, button styling

shashaxdu committed:
Merge remote-tracking branch 'origin/relativeStyling' into relativeStyling

Conflicts:
	css/IndividualtoInfinity.css

shashaxdu committed:
Fixed issues from last merge

shashaxdu commented on Avatar.html:
```diff
@@ -141,7 +144,7 @@ <h3 id="Badgegalaxy">  </h3>
 					<div class="col-xs-4">
 						<div class="container-outer">
 							<div class="container-inner hair">
-								<img src="img/avatar/Hair_01.png" " id="001">
```
I keep getting an error message of "Special characters must be escaped" 

<img width="1041" alt="screen shot 2017-09-30 at 6 10 43 pm" src="https://user-images.githubusercontent.com/13709857/31050845-6e129634-a60c-11e7-85b6-33f7905c4268.png">

Ibexian replied:
You have a " in front of id="001" - if you remove it you should be fine

shashaxdu commented:
1. So here's what the parallax should look like: 

![gif_parallax_earth](https://user-images.githubusercontent.com/13709857/31051073-b959a5d6-a613-11e7-8d85-e8547151f075.gif)
![gif_parallax_homepage](https://user-images.githubusercontent.com/13709857/31051074-b95b107e-a613-11e7-80fd-912caa91d3cc.gif)

2. On the homepage, the Avatar/Infinity icons aren't getting bigger for me with mouse over. Is it working for you?

On the next topic, there are a bunch of issues with the two Avatar creation pages and the code is a bit too complicated for me to figure out. I'll work on documenting the issues. 

In the meantime, I created a prototype for how the mobile experience should look and flow for the Avatar pages: https://youtu.be/7UVySU0e0UY 
Is this possible within the current framework for the site? 

If everything looks good to you, I can merge all the changes into Master.

Thank you!!!
